x/superfluid/keeper/stake_test.go

```diff
@@ -1188,6 +1192,173 @@ func (suite *KeeperTestSuite) TestPartialSuperfluidUndelegateToConcentratedPosit
+			positionId:                  2,
+			lowerTick:                   cltypes.MinInitializedTick,
+			upperTick:                   cltypes.MaxTick,
+			superfluidNotEnabledOnDenom: true,
+			expectedError:               errorsmod.Wrapf(types.ErrNonSuperfluidAsset, "denom: %s", "cl/pool/2"),
+		},
+		{
+			name:          "position lower tick is not full range",
+			positionId:    2,
+			lowerTick:     cltypes.MinInitializedTick + 1,
+			upperTick:     cltypes.MaxTick,
+			expectedError: cltypes.PositionNotFullRangeError{PositionId: 2, LowerTick: cltypes.MinInitializedTick + 1, UpperTick: cltypes.MaxTick},
+		},
+		{
+			name:          "position upper tick is not full range",
+			positionId:    2,
+			lowerTick:     cltypes.MinInitializedTick,
+			upperTick:     cltypes.MaxTick - 1,
+			expectedError: cltypes.PositionNotFullRangeError{PositionId: 2, LowerTick: cltypes.MinInitializedTick, UpperTick: cltypes.MaxTick - 1},
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			s.SetupTest()
+
+			// Set up testing environment.
+			s.FundAcc(s.TestAccs[1], defaultFunds)
+			valAddr, clPoolId, err := s.SetupConcentratedSuperfluidEnv(s.Ctx, s.TestAccs[1], tc.superfluidNotEnabledOnDenom)
+
+			// Create the position specified by the test case.
+			msg := &cltypes.MsgCreatePosition{
+				PoolId:          clPoolId,
+				Sender:          s.TestAccs[0].String(),
+				LowerTick:       tc.lowerTick,
+				UpperTick:       tc.upperTick,
+				TokensProvided:  defaultFunds,
+				TokenMinAmount0: sdk.ZeroInt(),
+				TokenMinAmount1: sdk.ZeroInt(),
+			}
+			msgServer := clkeeper.NewMsgServerImpl(s.App.ConcentratedLiquidityKeeper)
+			s.FundAcc(s.TestAccs[0], defaultFunds)
+			resp, err := msgServer.CreatePosition(sdk.WrapSDKContext(s.Ctx), msg)
+			s.Require().NoError(err)
+
+			// System under test
+			clLockId, err := s.App.SuperfluidKeeper.LockExistingFullRangePositionAndSFStake(s.Ctx, tc.positionId, s.TestAccs[0], valAddr.String())
+			if tc.expectedError != nil {
+				s.Require().Error(err)
+				s.Require().Equal(tc.expectedError.Error(), err.Error())
+				return
+			} else {
+				s.Require().NoError(err)
+				clLock, err := s.App.LockupKeeper.GetLockByID(s.Ctx, clLockId)
+				s.Require().NoError(err)
+				s.Require().Equal(resp.LiquidityCreated.TruncateInt(), clLock.Coins[0].Amount)
+
+				// check synthetic lockup creation
+				synthLock, err := s.App.LockupKeeper.GetSyntheticLockup(s.Ctx, clLock.ID, keeper.StakingSyntheticDenom(clLock.Coins[0].Denom, valAddr.String()))
+				s.Require().NoError(err)
+				s.Require().Equal(synthLock.UnderlyingLockId, clLock.ID)
+				s.Require().Equal(synthLock.SynthDenom, keeper.StakingSyntheticDenom(clLock.Coins[0].Denom, valAddr.String()))
+				s.Require().Equal(synthLock.EndTime, time.Time{})
+
+				expAcc := types.NewSuperfluidIntermediaryAccount(clLock.Coins[0].Denom, valAddr.String(), 0)
+
+				// Check lockID connection with intermediary account
```

Comment:
can we also check the actual delegated amount and see if it matches the liquidity in the position?